libraries/botframework-schema/src/index.ts

```diff
@@ -17,939 +19,1122 @@ export * from './teams';
+     */
+    suggestedActions?: SuggestedActions;
+    /**
+     * Attachments
+     */
+    attachments?: Attachment[];
+    /**
+     * Represents the entities that were mentioned in the message.
+     */
+    entities?: Entity[];
+    /**
+     * Contains channel-specific content.
+     */
+    channelData?: any;
+    /**
+     * Indicates whether the recipient of a contactRelationUpdate was added or removed from the
+     * sender's contact list.
+     */
+    action?: string;
+    /**
+     * Contains the ID of the message to which this message is a reply.
+     */
+    replyToId?: string;
+    /**
+     * A descriptive label for the activity.
+     */
+    label: string;
+    /**
+     * The type of the activity's value object.
+     */
+    valueType: string;
+    /**
+     * A value that is associated with the activity.
+     */
+    value?: any;
+    /**
+     * The name of the operation associated with an invoke or event activity.
+     */
+    name?: ActivityEventNames | string;
+    /**
+     * A reference to another conversation or activity.
+     */
+    relatesTo?: ConversationReference;
+    /**
+     * The a code for endOfConversation activities that indicates why the conversation ended.
+     * Possible values include: 'unknown', 'completedSuccessfully', 'userCancelled', 'botTimedOut',
+     * 'botIssuedInvalidMessage', 'channelFailed'
+     */
+    code?: EndOfConversationCodes | string;
+    /**
+     * The time at which the activity should be considered to be "expired" and should not be
+     * presented to the recipient.
+     */
+    expiration?: Date;
+    /**
+     * The importance of the activity. Possible values include: 'low', 'normal', 'high'
+     */
+    importance?: ActivityImportance | string;
+    /**
+     * A delivery hint to signal to the recipient alternate delivery paths for the activity.
+     * The default delivery mode is "default". Possible values include: 'normal', 'notification', 'expectReplies', 'ephemeral'
+     */
+    deliveryMode?: DeliveryModes | string;
+    /**
+     * List of phrases and references that speech and language priming systems should listen for
+     */
+    listenFor: string[];
+    /**
+     * The collection of text fragments to highlight when the activity contains a ReplyToId value.
+     */
+    textHighlights?: TextHighlight[];
+    /**
+     * An optional programmatic action accompanying this request
+     */
+    semanticAction?: SemanticAction;
+}
+
+export const assertActivity: Assertion<Activity> = (value, path) => {
+    assert.unsafe.castObjectAs<Activity>(value, path);
+    assert.string(value.type, path.concat('type'));
+    assert.maybeString(value.id, path.concat('id'));
+    assert.maybeDate(value.timestamp, path.concat('timestamp'));
+    assert.maybeDate(value.localTimestamp, path.concat('localTimestamp'));
+    assert.string(value.localTimezone, path.concat('localTimezone'));
+    assert.string(value.callerId, path.concat('calledId'));
```

Comment:
Typo here, and callerId can be `null` or `undefined`. Depending on the bot implementation. Currently callerId is only populated by the `BotFrameworkAdapter`.

Reply:
Updated the interface and fixed the typo.